frontend/src/components/homepage/homepage.tsx

```diff
@@ -13,24 +14,42 @@ const Homepage: React.FC = () => {
   const dispatch = useDispatch();
   const hasPodcasts = Boolean(podcasts.length);
 
+  const [filter, setFilter] = useState<Array<Podcast>>(podcasts);
+
   useEffect(() => {
     dispatch(homepageActions.loadPodcasts());
   }, []);
 
+  const handleChange = (searchValue: SearchPayload): void => {
+    const { search } = searchValue;
+    const filtered = filterPodcasts(search);
+    const payload = !search ? podcasts : filtered;
+    setFilter(payload);
+  };
+
+  const filterPodcasts = (search: string): Array<Podcast> => {
```

Comment:
move to `frontend/src/components/helpers/helpers`

```
const getFilteredPodcasts = (podcasts: Podcast[], searchValue: string): Podcast[] => {
  return podcasts.filter((it) => it.name.toLowerCase().includes(filterValue.toLowerCase())
}
```

then, inside component 
```
const filteredPodcasts = getFilteredPodcasts(podcasts, searchValue);
```

then, in markup

```
<PodcastList podcasts={filteredPodcasts} />
```